SeQuant/domain/mbpt/convention.cpp

```diff
@@ -62,118 +28,138 @@ auto qndecorate(qns qn, std::wstring_view label) {
-    // introduced in MPQC for GF, CT-F12, and other ad hoc uses
-    IndexSpace::register_instance(declab(L"x"), IndexSpace::all_active, qnattr,
-                                  do_not_throw);
-    // introduced here
-    IndexSpace::register_instance(declab(L"γ"),
-                                  IndexSpace::complete_inactive_unoccupied,
-                                  qnattr, do_not_throw);
-    // e.g. see DOI 10.1063/5.0067511
-    IndexSpace::register_instance(declab(L"u"), IndexSpace::active, qnattr,
-                                  do_not_throw);
-    // DOI 10.1063/5.0067511 uses I,J,K... and A,B,C... for these
-    // although QCiFS uses capital letters for spin-free indices, using I/A this
-    // way seems preferable
-    IndexSpace::register_instance(
-        declab(L"I"), IndexSpace::active_maybe_occupied, qnattr, do_not_throw);
-    IndexSpace::register_instance(declab(L"A"),
-                                  IndexSpace::active_maybe_unoccupied, qnattr,
-                                  do_not_throw);
-    // introduced here
-    IndexSpace::register_instance(declab(L"M"), IndexSpace::maybe_occupied,
-                                  qnattr, do_not_throw);
-    IndexSpace::register_instance(declab(L"E"), IndexSpace::maybe_unoccupied,
-                                  qnattr, do_not_throw);
-    IndexSpace::register_instance(declab(L"Δ"),
-                                  IndexSpace::complete_maybe_unoccupied, qnattr,
-                                  do_not_throw);
-  }
+  // add irreducible representations first
+  IndexSpace frozen(L"o",{0b00001});
+  standard_reference_registry.add(frozen);
+
+  IndexSpace active_occ(L"i", {0b00010});
+  standard_reference_registry.add(active_occ);
+
+  IndexSpace active_uocc(L"a", {0b00100});
+  standard_reference_registry.add(active_uocc);
+
+  IndexSpace inactive_uocc(L"g", {0b01000});
+  standard_reference_registry.add(inactive_uocc);
+
+  IndexSpace other_uocc(L"α'", {0b10000});
+  standard_reference_registry.add(other_uocc);
+
+  IndexSpace occupied(L"m", frozen.type().unIon(active_occ.attr()).to_int32());
+  standard_reference_registry.add(occupied);
+
+  IndexSpace unoccupied(L"e",active_uocc.type().unIon(inactive_uocc.type()).to_int32());
+  standard_reference_registry.add(unoccupied);
+
+  IndexSpace active (L"x", active_occ.type().unIon(active_uocc.type()).to_int32());
+  standard_reference_registry.add(active);
+
+  IndexSpace complete_unoccupied(L"α",unoccupied.type().unIon(other_uocc.type()).to_int32());
+  standard_reference_registry.add(complete_unoccupied);
+
+  IndexSpace all(L"p",unoccupied.type().unIon(occupied.type()).to_int32());
+  standard_reference_registry.add(all);
+
+  IndexSpace complete(L"κ",complete_unoccupied.type().unIon(occupied.type()).to_int32());
+  standard_reference_registry.add(complete);
+
+  //only neccessary for some Operator definitions
+  standard_reference_registry.assign_density_occupied(L"m");
```

Comment:
Maybe an overload for the `assign` functions would be nice that takes an index space object and then internally resorts to adding the index space and associating its label with the respective group. That would allow to configure such spaces with a single function call instead of two.